test/Streamly/Test/Data/Parser.hs

```diff
@@ -1129,6 +1132,29 @@ takeStartBy_ =
             where
                 predicate = odd
                 parser = P.takeStartBy_ predicate FL.toList
+
+-- Scientific notation
+scientificExpFP :: Property
+scientificExpFP =  forAll (chooseDouble (-99.99e-12, 1234.4567e+234)) $ \ls ->
+                case runIdentity $ S.parse parser (S.fromList (show ls)) of
+                     Right val -> property (val == show ls)
+                     Left _ -> property False
+
+                    where
+                    formatter = Scientific.formatScientific Scientific.Exponent Nothing
+                    parser = formatter <$> Parser.scientific
+
+-- Standard decimal notation.
+scientificFixFP :: Property
+scientificFixFP =  forAll (chooseDouble (-0.00099, 123445.67998)) $ \ls ->
+                case runIdentity $ S.parse parser (S.fromList (show ls)) of
```

Comment:
Can you explain why you chose this range?

Reply:
cover the value range of negative and positive with a negative and positive exponent.